tests/testlib/s2n_test_certs.c

```diff
@@ -76,3 +76,23 @@ int s2n_read_test_pem_and_len(const char *pem_path, uint8_t *pem_out, uint32_t *
     return 0;
 }
 
+S2N_RESULT s2n_test_cert_chain_from_pem(struct s2n_connection *conn, const char *pem_path,
+        struct s2n_stuffer *cert_chain_stuffer) {
+    RESULT_ENSURE_REF(cert_chain_stuffer);
+    RESULT_GUARD_POSIX(s2n_stuffer_growable_alloc(cert_chain_stuffer, 4096));
+
+    uint8_t cert_chain_pem[S2N_MAX_TEST_PEM_SIZE] = { 0 };
+    uint32_t cert_chain_pem_len = 0;
+    RESULT_GUARD_POSIX(s2n_read_test_pem_and_len(pem_path, cert_chain_pem, &cert_chain_pem_len, S2N_MAX_TEST_PEM_SIZE));
+
+    DEFER_CLEANUP(struct s2n_cert_chain_and_key *chain_and_key = s2n_cert_chain_and_key_new(),
+            s2n_cert_chain_and_key_ptr_free);
+    RESULT_GUARD_POSIX(s2n_cert_chain_and_key_load_public_pem_bytes(chain_and_key, cert_chain_pem, cert_chain_pem_len));
+
+    RESULT_GUARD_POSIX(s2n_send_cert_chain(conn, cert_chain_stuffer, chain_and_key));
+
+    /* Skip the cert chain length */
+    RESULT_GUARD_POSIX(s2n_stuffer_skip_read(cert_chain_stuffer, 3));
+
```

Comment:
Do you think this will cause problems for anyone if they call reread? This is just for testing so it's not super important, but we could also just copy to another stuffer without the length.

Reply:
Right okay, that makes sense. I changed this to write to a temporary stuffer and then copy everything except for the certificate chain length to the return stuffer.